MekHQ/src/mekhq/gui/dialog/PartsReportDialog.java

```diff
@@ -127,6 +130,43 @@ public void actionPerformed(ActionEvent e) {
                 refreshOverviewSpecificPart(row, piu, partToBuy);
             }
         };
+
+        Action sell = new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                int row = Integer.parseInt(e.getActionCommand());
+                PartInUse piu = overviewPartsModel.getPartInUse(row);
+                campaign.getWarehouse().getSpareParts().stream().filter(p ->
+                    Objects.equals(p.getName(), piu.getName()))
+                    .findFirst()
+                    .ifPresent(p -> campaign.getQuartermaster().sellPart(p, 1));
+                refreshOverviewPartsInUse();
+            }
+        };
+
+        Action sellInBulk = new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                int row = Integer.parseInt(e.getActionCommand());
+                PartInUse piu = overviewPartsModel.getPartInUse(row);
+                campaign.getWarehouse().getSpareParts().stream().filter(p ->
+                                Objects.equals(p.getName(), piu.getName()))
+                        .findFirst()
+                        .ifPresent(p -> {
+                            int quantity = 1;
+                            PopupValueChoiceDialog pcd = new PopupValueChoiceDialog(gui.getFrame(), true,
+                                    "Sell how many " + p.getName(), quantity, 1, CampaignGUI.MAX_QUANTITY_SPINNER);
+                            pcd.setVisible(true);
```

Comment:
As an illustration of my earlier point, I've reached this and I've no idea what `pcd` stands for. Eventually I found it on line 157, but if it were called `popupChoiceDialog` I would have immediately known what the variable was for.

Reply:
Done in cdc6ddb